syntax = "proto3";

message Message {
  string sender = 1;
  string content = 2;
  
}

service Chat {
  rpc SendMessage(MessageRequest) returns (MessageResponse);
  rpc ReceiveMessages(ClientInfo) returns (stream Message);
}

message MessageRequest {
  string sender = 1;
  string content = 2;

}

message MessageResponse {
  string status = 1;
}

message ClientInfo {
  string username = 1;
  string ipAddress = 2;
  int32 port = 3;
  repeated string receivedMessages = 4;
}
